aws-logs-destination/src/main/java/software/amazon/logs/destination/DeleteHandler.java

```diff
@@ -0,0 +1,50 @@
+package software.amazon.logs.destination;
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DeleteDestinationRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DeleteDestinationResponse;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class DeleteHandler extends BaseHandlerStd {
+
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request, final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient, final Logger logger) {
+
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+
+        ProgressEvent<ResourceModel, CallbackContext> progressEvent = ProgressEvent.progress(model, callbackContext)
+                .then(progress -> proxy.initiate("AWS-Logs-Destination::Delete", proxyClient, model, callbackContext)
+                        .translateToServiceRequest(Translator::translateToDeleteRequest)
+                        .makeServiceCall(this::deleteResource)
+                        .success());
+
+        logger.log(String.format("%s resource with name %s has been successfully deleted", ResourceModel.TYPE_NAME,
+                model.getDestinationName()));
```

Comment:
This will be logged _before_ the resource was deleted. The code inside 25-28 is asynchronous